snuba/web/query.py

```diff
@@ -63,13 +64,84 @@ def visit_join_clause(self, node: JoinClause[Entity]) -> bool:
         return node.left_node.accept(self) or node.right_node.accept(self)
 
 
+@with_span()
+def run_query_with_quota_control(
+    dataset: Dataset,
+    request: Request,
+    timer: Timer,
+    robust: bool = False,
+    concurrent_queries_gauge: Optional[Gauge] = None,
+) -> QueryResult:
+    def exec_query(
+        dataset: Dataset,
+        request: Request,
+        timer: Timer,
+        query_metadata: SnubaQueryMetadata,
+        robust: bool = False,
+        concurrent_queries_gauge: Optional[Gauge] = None,
+    ) -> QueryResult:
+        try:
+            with dataset.get_quota_control_policy().acquire(request) as stats:
```

Comment:
This call is a noop if the setting is disabled, so I think you could modify `parse_and_run_query` to have this call in it, and then there's no need to have both `run_query_with_quota_control` and `parse_and_run_query`.